### PR TITLE
Drop support for Python 3.9 and 3.10 and add 3.12

### PR DESCRIPTION
**Pull request by Mark90**
* Drop Python 3.9 and 3.10 support, remove backwards compatibility code
* Add Python 3.12 support
* Used pyupgrade to rewrite `List[A]` to `list[A]`, `Union[A, B]` to `A | B`, etc
* Fixed type annotations in `orchestrator/domain/lifecycle.py` and `orchestrator/api/models.py`

### CODE REVIEW

File: orchestrator/cli/domain_gen_helpers/helpers.py [hunk 1]
```diff
@@ -27,29 +27,29 @@ def generic_mapper(prop_name: str, model_diffs: Dict[str, Dict[str, Set[str]]])
     return {k: {val[1] for val in v} for k, v in grouped}
 
 
-def map_create_fixed_inputs(model_diffs: Dict[str, Dict[str, Set[str]]]) -> Dict[str, Set[str]]:
+def map_create_fixed_inputs(model_diffs: dict[str, dict[str, set[str]]]) -> dict[str, set[str]]:
```
Comment (mrijk) [inline orchestrator/cli/domain_gen_helpers/helpers.py:30]: Couple of type aliases might someday help to make this typing a bit more readable.

**Review by mrijk — APPROVED** [review]
LGTM

File: orchestrator/cli/domain_gen_helpers/helpers.py (hunk 1 above)
Comment (Mark90) [reply orchestrator/cli/domain_gen_helpers/helpers.py:30]: Agree, I don't really know what is passed along here

**Review by tjeerddie — APPROVED** [review]

File: orchestrator/api/api_v1/endpoints/product_blocks.py [hunk 2]
```diff
@@ -29,8 +28,8 @@
 router = APIRouter()
 
 
-@router.get("/", response_model=List[ProductBlockSchema])
-def fetch() -> List[ProductBlockTable]:
+@router.get("/", response_model=list[ProductBlockSchema])
+def fetch() -> list[ProductBlockTable]:
```
[the remark this answers is not in the document]
Comment (Mark90) [reply orchestrator/api/api_v1/endpoints/product_blocks.py:32]: Will do this in a separate branch based off this one. It requires a bit more than just omitting the response_model and don't want to hide it in a bulk of changes

**Review by pboers1988 — APPROVED** [review]
LGTM

**Pull request merged**